test_semver.py

```diff
@@ -497,3 +497,15 @@ def test_immutable_unknown_attribute(version):
 def test_version_info_should_be_iterable(version):
     assert tuple(version) == (version.major, version.minor, version.patch,
                               version.prerelease, version.build)
+
+
+@pytest.mark.parametrize("version, index, expected", [
+    ("1.2.3-rc.0+build.0", 0, 1),
+    ("1.2.3-rc.0+build.0", 1, 2),
+    ("1.2.3-rc.0+build.0", 2, 3),
+    ("1.2.3-rc.0+build.0", 3, "rc.0"),
+    ("1.2.3-rc.0+build.0", 4, "build.0"),
+])
+def test_version_info_should_be_accessed_with_index(version, index, expected):
+    version_info = VersionInfo.parse(version)
+    assert version_info[index] == expected
```

Comment:
I would add more tests for corner case scenarios - for instance - what if are trying to get index of prerelease part but `VersionInfo` does not have such field? e.g.
`assert VersionInfo.parse("1.2.3")[4] == None`

Also, please add tests for negative and too big index value:
`assert VersionInfo.parse("1.2.3-rc.0+build.0")[-1] == "build.0"` and
`with pytest.raises(IndexError): VersionInfo.parse("1.2.3-rc.0+build.0")[5]`

Reply:
I'll add test vectors for these cases.

Reply:
> I would add more tests for corner case scenarios - for instance - what if are trying to get index of prerelease part but `VersionInfo` does not have such field? e.g.
> `assert VersionInfo.parse("1.2.3")[4] == None`
> 
> Also, please add tests for negative and too big index value:
> `assert VersionInfo.parse("1.2.3-rc.0+build.0")[-1] == "build.0"` and
> `with pytest.raises(IndexError): VersionInfo.parse("1.2.3-rc.0+build.0")[5]`

That's a good idea! May I suggest to add the tests for a negative index value as a separate test function?
I think, that makes it a bit easier as you don't have to change `test_version_info_should_be_accessed_with_index` (as test function should be short).